semantic-python/src/Language/Python/Core.hs

```diff
@@ -194,7 +194,19 @@ instance Compile Py.Call where
     locate it (func $$* args) & cc
   compile it _ _ = fail ("can't compile Call node with generator expression: " <> show it)
 
-instance Compile Py.ClassDefinition
+instance Compile Py.ClassDefinition where
+  compile it@Py.ClassDefinition { body = pybody, name = Py.Identifier _ann (Name -> n) } cc next = do
+    let buildTypeCall _ = do
+          bindings <- asks @Bindings (toList . unBindings)
+          let buildName n = (n, pure n)
+          let contents = record . fmap buildName $ bindings
+          pure (pure (Name "type") $$ Core.string (coerce n) $$ pure "object" $$ contents)
+
+    body <- compile pybody buildTypeCall next
+    let assignClass = Name.named' n :<- body
+    let continuing = fmap (locate it . (assignClass >>>=))
+    continuing (local (def n) (cc next))
```

Comment:
We use `local` again here because we want to record the name of this class so that it can be exported.